### PR TITLE
cmd/worker: migrate to lib/log

### PR DESCRIPTION
**Pull request by bobheadxi**
I confused myself again with all the different workers! I am spinning this off into a separate PR since I realized it is not related to https://github.com/sourcegraph/sourcegraph/pull/34155 

This PR migrates the following to `lib/log`:

- the `Job` interface
- `cmd/worker`
- `enterprise/cmd/worker`

The scope of this PR is to simply update the interface and use the loggers immediately where appropriate, to give a starting point for eventually migrating everything.

Many callsites use `log15.Root()` for `internal/observation`, which will be tackled in https://github.com/sourcegraph/sourcegraph/pull/34456

## Test plan

`sg start`:

image

Empty attributes is slightly annoying, TBD if we should address - a potential workaround is detect in `zapAdapter` whether a first field is provided, and only add the Attributes namespace if so

### CODE REVIEW

File: cmd/worker/job/job.go [hunk 1]
```diff
@@ -5,11 +5,15 @@ import (
...
 	"github.com/sourcegraph/sourcegraph/internal/env"
 	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/lib/log"
 )
 
 // Job creates configuration struct and background routine instances to be run
 // as part of the worker process.
 type Job interface {
+	// Description renders a brief overview of what this job does and handles.
+	Description() string
```
Comment (efritz) [inline cmd/worker/job/job.go:15]: <3

File: cmd/worker/shared/main.go [hunk 2]
```diff
@@ -107,6 +112,8 @@ func Start(additionalJobs map[string]job.Job, registerEnterpriseMigrations func(
 	close(ready)
 
 	goroutine.MonitorBackgroundRoutines(context.Background(), allRoutines...)
```
Comment (efritz) [inline cmd/worker/shared/main.go:114]: The new return here makes it seem like it's reachable when it's just to appease the type checker. Could we add a note here that this call blocks while the app is live?

File: cmd/worker/shared/main.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ func Start(additionalJobs map[string]job.Job, registerEnterpriseMigrations func(
 
 	// Set up Google Cloud Profiler when running in Cloud
 	if err := profiler.Init(); err != nil {
-		log.Fatalf("Failed to start profiler: %v", err)
+		return errors.Wrap(err, "Failed to start profiler")
```
Comment (efritz) [inline cmd/worker/shared/main.go:67]: Do we wrap with lowercase strings or title case strings by convention?

Comment (bobheadxi) [reply cmd/worker/shared/main.go:67]: I've seen both for sure, and I'm not seeing any linter warnings on these

Comment (unknwon) [reply cmd/worker/shared/main.go]: Linter does complain about having error messages start with non-lowercase (except coined terms), it's just linter doesn't support parse our `errors.XYZ`.

**Review by unknwon — APPROVED** [review]

File: cmd/worker/shared/main.go (hunk 3 above)
Comment (bobheadxi) [reply cmd/worker/shared/main.go]: I don't think there's _too_ much of a benefit for standardizing on lower/title-case for error messages so I'm just going to leave this as is

**Pull request merged**